src/main/cpp/subsys/BallHopper.cpp

```diff
@@ -0,0 +1,38 @@
+//====================================================================================================================================================
+// Copyright 2020 Lake Orion Robotics FIRST Team 302
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE
+// OR OTHER DEALINGS IN THE SOFTWARE.
+//====================================================================================================================================================
+
+//C++ Includes
+#include <memory>
+
+//Team 302 Includes
+#include <subsys/BallHopper.h>
+#include <hw/interfaces/IDragonMotorController.h>
+#include <hw/DragonDigitalInput.h>
+
+using namespace std;
+
+BallHopper::BallHopper
+(
+    shared_ptr<IDragonMotorController>  motorController,
+    shared_ptr<DragonDigitalInput>      ballDetection
+) : Mech1IndMotor( MechanismTypes::MECHANISM_TYPE::BALL_HOPPER, string( "ballhopper.xml"), string("BallHopperNT"), motorController)
```

Comment:
its preferable to do

) : Mech1IndMotor(....), m_bannerSensor(ballDetection)

Also, it would be better to not refer to it as a bannersensor rather what it is doing (if they change the sensor type for detecting balls, you don't want the variable to be misleading)